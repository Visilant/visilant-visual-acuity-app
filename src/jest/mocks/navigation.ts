import { RootStackParamList } from '@screens/stack'

export const mockedNavigation = {
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  goBack: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  setParams: jest.fn(),
  dispatch: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getId: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  setOptions: jest.fn()
}

export const getMockedRoute = (routeName: keyof RootStackParamList) => ({
  key: '',
  name: routeName
})
