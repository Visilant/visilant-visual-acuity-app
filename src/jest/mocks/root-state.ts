import { RootState } from '@store'

const mockRootState: RootState = {
  _persist: { version: 1, rehydrated: false },
  examination: {
    entities: [],
    activeExamination: undefined,
    deepLinkParams: undefined
  },
  auth: {
    token: 'mock-token',
    refresh_token: 'mock-refresh-token',
    _persist: {
      version: 0,
      rehydrated: false
    }
  },
  settings: { showDashboardInstructions: true, locale: 'en-EN' },
  snackbar: { alerts: [] },
  network: {
    isQueuePaused: false,
    actionQueue: [],
    isConnected: null
  }
}

export const getMockState = () => JSON.parse(JSON.stringify(mockRootState))
