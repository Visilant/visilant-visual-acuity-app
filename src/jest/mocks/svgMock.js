export default 'svg'
export const ReactComponent = 'View'
