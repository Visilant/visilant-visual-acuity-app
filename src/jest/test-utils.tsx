import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { fireEvent, render, RenderOptions, screen } from '@testing-library/react-native'
import { WrapperProps } from './types'
import { ThemeProvider } from 'styled-components/native'
import { appTheme } from '@components/theme/theme'
import { LocalizationProvider } from '@localizations/Provider'
import { store } from '@store'

const Providers: React.FunctionComponent<WrapperProps> = ({ children }) => {
  return (
    <ReduxProvider store={store}>
      <ThemeProvider theme={appTheme}>
        <LocalizationProvider>{children}</LocalizationProvider>
      </ThemeProvider>
    </ReduxProvider>
  )
}

const customRender = <T,>(
  ui: React.ReactElement<unknown, string | React.JSXElementConstructor<T>>,
  options?: RenderOptions
) => {
  return render(ui, {
    wrapper: Providers,
    ...options
  })
}

export { fireEvent, customRender as render, screen }
