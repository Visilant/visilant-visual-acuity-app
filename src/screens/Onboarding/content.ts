export const stepContents = [
  {
    animation: require('@assets/animations/onboarding-1.gif'),
    title: 'onboarding.place.title',
    text: 'onboarding.place.text'
  },
  {
    animation: require('@assets/animations/onboarding-2.gif'),
    title: 'onboarding.position.title',
    text: 'onboarding.position.text'
  },
  {
    animation: require('@assets/animations/onboarding-3.gif'),
    title: 'onboarding.instruct.title',
    text: 'onboarding.instruct.text'
  },
  {
    animation: require('@assets/animations/onboarding-4.gif'),
    title: 'onboarding.test.title',
    text: 'onboarding.test.text'
  }
]
