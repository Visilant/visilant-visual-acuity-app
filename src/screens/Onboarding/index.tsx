import { Stack } from '@screens/stack'
import React from 'react'
import Onboarding from './Onboarding'

const Screen = () => {
  return (
    <>
      <Stack.Screen name="Onboarding" component={Onboarding} options={{ title: 'Onboarding' }} />
    </>
  )
}

export default Screen
