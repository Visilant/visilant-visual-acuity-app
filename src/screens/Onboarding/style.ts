import { Button, Card } from '@components'
import { Image, View } from 'react-native'
import styled from 'styled-components/native'

export const Header = styled(View)`
  margin-bottom: 16px;
`
export const Animation = styled(Image)`
  width: 100%;
  height: 100%;
`
export const AnimationCard = styled(Card)`
  padding: 0;
  overflow: hidden;
`
export const Captions = styled(View)`
  position: 'relative';
  min-height: 80px;
`
export const CaptionTitle = styled(View)`
  margin-top: -10px;
  margin-bottom: 20px;
`
export const StartTestButton = styled(Button)`
  margin-top: auto;
  flex-grow: 0;
`
