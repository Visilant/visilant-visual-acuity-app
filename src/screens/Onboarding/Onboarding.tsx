import { Body, SafeView, Screen, Text } from '@components'
import { Carousel } from '@components/Carousel'
import { CarouselScrollState } from '@components/Carousel/types'
import { FadeInOutView } from '@components/FadeInOutView'
import { TextStyles } from '@components/Text'
import { useFullWidth } from '@hooks/useFullWidth'
import React, { useState } from 'react'
import { Image, View } from 'react-native'
import { RootStackScreenProps } from '@screens/stack'
import { stepContents } from './content'
import { Animation, AnimationCard, CaptionTitle, Captions, Header, StartTestButton } from './style'
import { withErrorBoundary } from '@hocs/withErrorBoundary'

const Onboarding = ({ navigation }: RootStackScreenProps<'Onboarding'>) => {
  const [page, setPage] = useState(0)
  const [scrollState, setScrollState] = useState(CarouselScrollState.idle)
  const imageWidth = useFullWidth()

  return (
    <Screen noBg>
      <SafeView compact>
        <Header>
          <Image source={require('@assets/logo.png')} />
        </Header>
        <Body fullHeight>
          <Carousel
            height={imageWidth}
            onPageChange={index => setPage(index)}
            onScrollStateChanged={state => setScrollState(state)}
          >
            {stepContents.map((step, i) => (
              <View key={step.title}>
                <AnimationCard>
                  <Animation
                    source={stepContents[i].animation}
                    style={{ width: imageWidth, height: imageWidth }}
                  />
                </AnimationCard>
              </View>
            ))}
          </Carousel>

          <Captions>
            {stepContents.map((step, i) => (
              <FadeInOutView
                absolute
                key={step.title}
                visible={page === i && scrollState === CarouselScrollState.idle}
              >
                <CaptionTitle>
                  <Text namespace="screens" text={step.title} textStyle={TextStyles.Title} />
                </CaptionTitle>
                <Text namespace="screens" text={step.text} textStyle={TextStyles.Paragraph} />
              </FadeInOutView>
            ))}
          </Captions>

          <StartTestButton
            namespace="screens"
            text="onboarding.button"
            capitalize={true}
            onPress={() => navigation.navigate('Test', {})}
          />
        </Body>
      </SafeView>
    </Screen>
  )
}

export default withErrorBoundary(Onboarding)
