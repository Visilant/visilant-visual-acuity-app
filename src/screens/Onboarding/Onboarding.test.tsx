import { mockedNavigation } from '@jest/mocks/navigation'
import { fireEvent, render, screen } from '@jest/test-utils'
import Onboarding from './Onboarding'

describe('Onboarding screen', () => {
  test('matches the snapshot', async () => {
    const onboarding = render(
      <Onboarding navigation={mockedNavigation} route={{ key: '', name: 'Onboarding' }} />
    )

    expect(onboarding).toMatchSnapshot()
  })

  test('the "start test" button navigates away', () => {
    render(<Onboarding navigation={mockedNavigation} route={{ key: '', name: 'Onboarding' }} />)

    // eslint-disable-next-line quotes
    const btn = screen.getByText("LET'S START")
    fireEvent(btn, 'press')

    expect(mockedNavigation.navigate).toBeCalledTimes(1)
  })

  test('renders starter instructions', async () => {
    render(<Onboarding navigation={mockedNavigation} route={{ key: '', name: 'Onboarding' }} />)

    screen.getByText('Find the place')
    screen.getByText('Make sure the screen is easy to see in an indoor or shaded environment.')
  })
})
