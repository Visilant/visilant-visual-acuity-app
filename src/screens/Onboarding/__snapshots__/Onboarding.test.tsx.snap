// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding screen matches the snapshot 1`] = `
<View
  noBg={true}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    compact={true}
    insets={
      {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 0,
      }
    }
  >
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "flexDirection": "column",
          "flexGrow": 1,
          "gap": 35,
          "justifyContent": "flex-start",
          "paddingBottom": 35,
          "paddingTop": 30,
        }
      }
      style={
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View>
        <RNCViewPager
          initialPage={0}
          layoutDirection="ltr"
          onMoveShouldSetResponderCapture={[Function]}
          onPageScroll={[Function]}
          onPageScrollStateChanged={[Function]}
          onPageSelected={[Function]}
          pageMargin={30}
          style={
            {
              "height": 690,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                [
                  undefined,
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "rgba(255,255,255,0.5)",
                    "borderBottomLeftRadius": 25,
                    "borderBottomRightRadius": 25,
                    "borderColor": "#e6eef4",
                    "borderStyle": "solid",
                    "borderTopLeftRadius": 25,
                    "borderTopRightRadius": 25,
                    "borderWidth": 1,
                    "gap": 30,
                    "overflow": "hidden",
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    [
                      {
                        "height": "100%",
                        "width": "100%",
                      },
                      {
                        "height": 690,
                        "width": 690,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                [
                  undefined,
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "rgba(255,255,255,0.5)",
                    "borderBottomLeftRadius": 25,
                    "borderBottomRightRadius": 25,
                    "borderColor": "#e6eef4",
                    "borderStyle": "solid",
                    "borderTopLeftRadius": 25,
                    "borderTopRightRadius": 25,
                    "borderWidth": 1,
                    "gap": 30,
                    "overflow": "hidden",
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    [
                      {
                        "height": "100%",
                        "width": "100%",
                      },
                      {
                        "height": 690,
                        "width": 690,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                [
                  undefined,
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "rgba(255,255,255,0.5)",
                    "borderBottomLeftRadius": 25,
                    "borderBottomRightRadius": 25,
                    "borderColor": "#e6eef4",
                    "borderStyle": "solid",
                    "borderTopLeftRadius": 25,
                    "borderTopRightRadius": 25,
                    "borderWidth": 1,
                    "gap": 30,
                    "overflow": "hidden",
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    [
                      {
                        "height": "100%",
                        "width": "100%",
                      },
                      {
                        "height": 690,
                        "width": 690,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                [
                  undefined,
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "rgba(255,255,255,0.5)",
                    "borderBottomLeftRadius": 25,
                    "borderBottomRightRadius": 25,
                    "borderColor": "#e6eef4",
                    "borderStyle": "solid",
                    "borderTopLeftRadius": 25,
                    "borderTopRightRadius": 25,
                    "borderWidth": 1,
                    "gap": 30,
                    "overflow": "hidden",
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    [
                      {
                        "height": "100%",
                        "width": "100%",
                      },
                      {
                        "height": 690,
                        "width": 690,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </RNCViewPager>
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 10,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessibilityLabel="Jump to page 1"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selected={true}
            style={
              {
                "backgroundColor": "#EF4823",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 10,
                "width": 10,
              }
            }
          />
          <View
            accessibilityLabel="Jump to page 2"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selected={false}
            style={
              {
                "backgroundColor": "#E6EEF4",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 10,
                "width": 10,
              }
            }
          />
          <View
            accessibilityLabel="Jump to page 3"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selected={false}
            style={
              {
                "backgroundColor": "#E6EEF4",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 10,
                "width": 10,
              }
            }
          />
          <View
            accessibilityLabel="Jump to page 4"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selected={false}
            style={
              {
                "backgroundColor": "#E6EEF4",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 10,
                "width": 10,
              }
            }
          />
        </View>
        <View
          style={
            {
              "minHeight": 80,
              "position": "'relative'",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 0,
                "position": "absolute",
                "transform": [],
              }
            }
          >
            <View
              style={
                {
                  "marginBottom": 20,
                  "marginTop": -10,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#23255A",
                      "fontFamily": "Caprasimo",
                      "fontSize": 36,
                      "lineHeight": 38,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    },
                    "
  ",
                    [Function],
                    "
  font-family: Caprasimo;
  font-size: 36px;
  line-height: 38px;
",
                  ]
                }
              >
                Find the place
              </Text>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#23255A",
                    "fontFamily": "Libre-Franklin-ExtraLight",
                    "fontSize": 18,
                    "lineHeight": 28,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                  "
  ",
                  [Function],
                  "
  font-size: 18px;
  font-family: Libre-Franklin-ExtraLight;
  line-height: 28px;
",
                ]
              }
            >
              Make sure the screen is easy to see in an indoor or shaded environment.
            </Text>
          </View>
          <View
            collapsable={false}
            style={
              {
                "opacity": 0,
                "position": "absolute",
                "transform": [],
              }
            }
          >
            <View
              style={
                {
                  "marginBottom": 20,
                  "marginTop": -10,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#23255A",
                      "fontFamily": "Caprasimo",
                      "fontSize": 36,
                      "lineHeight": 38,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    },
                    "
  ",
                    [Function],
                    "
  font-family: Caprasimo;
  font-size: 36px;
  line-height: 38px;
",
                  ]
                }
              >
                Set the position
              </Text>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#23255A",
                    "fontFamily": "Libre-Franklin-ExtraLight",
                    "fontSize": 18,
                    "lineHeight": 28,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                  "
  ",
                  [Function],
                  "
  font-size: 18px;
  font-family: Libre-Franklin-ExtraLight;
  line-height: 28px;
",
                ]
              }
            >
              Stand 1 or 2 meters away (based on the settings) from the patient.
            </Text>
          </View>
          <View
            collapsable={false}
            style={
              {
                "opacity": 0,
                "position": "absolute",
                "transform": [],
              }
            }
          >
            <View
              style={
                {
                  "marginBottom": 20,
                  "marginTop": -10,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#23255A",
                      "fontFamily": "Caprasimo",
                      "fontSize": 36,
                      "lineHeight": 38,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    },
                    "
  ",
                    [Function],
                    "
  font-family: Caprasimo;
  font-size: 36px;
  line-height: 38px;
",
                  ]
                }
              >
                Onboarding
              </Text>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#23255A",
                    "fontFamily": "Libre-Franklin-ExtraLight",
                    "fontSize": 18,
                    "lineHeight": 28,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                  "
  ",
                  [Function],
                  "
  font-size: 18px;
  font-family: Libre-Franklin-ExtraLight;
  line-height: 28px;
",
                ]
              }
            >
              Instruct the patient to point in the direction the E is facing.
            </Text>
          </View>
          <View
            collapsable={false}
            style={
              {
                "opacity": 0,
                "position": "absolute",
                "transform": [],
              }
            }
          >
            <View
              style={
                {
                  "marginBottom": 20,
                  "marginTop": -10,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#23255A",
                      "fontFamily": "Caprasimo",
                      "fontSize": 36,
                      "lineHeight": 38,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    },
                    "
  ",
                    [Function],
                    "
  font-family: Caprasimo;
  font-size: 36px;
  line-height: 38px;
",
                  ]
                }
              >
                Fill the test
              </Text>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#23255A",
                    "fontFamily": "Libre-Franklin-ExtraLight",
                    "fontSize": 18,
                    "lineHeight": 28,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                  "
  ",
                  [Function],
                  "
  font-size: 18px;
  font-family: Libre-Franklin-ExtraLight;
  line-height: 28px;
",
                ]
              }
            >
              Swipe your finger in the direction the patient points.
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          size={1}
          style={
            [
              {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#EF4823",
                "borderBottomLeftRadius": 31,
                "borderBottomRightRadius": 31,
                "borderTopLeftRadius": 31,
                "borderTopRightRadius": 31,
                "flexWrap": "nowrap",
                "height": 62,
                "justifyContent": "center",
                "overflow": "hidden",
                "paddingBottom": 12,
                "paddingLeft": 30,
                "paddingRight": 30,
                "paddingTop": 12,
              },
              {
                "flexGrow": 0,
                "marginTop": "auto",
              },
            ]
          }
          type="normal"
        >
          <View
            style={{}}
          />
          <Text
            props={
              {
                "size": 1,
                "type": "normal",
              }
            }
            style={
              [
                {
                  "color": "#E6EEF4",
                  "flexWrap": "nowrap",
                  "fontFamily": "Libre-Franklin-Bold",
                  "fontSize": 24,
                  "lineHeight": 38,
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "top": -1,
                },
                "
  position: relative;
  top: ",
                [Function],
                ";
  color: ",
                [Function],
                ";
  font-family: Libre-Franklin-Bold;

  ",
                [Function],
                ";
  text-align: center;
  flex-wrap: nowrap;
",
              ]
            }
          >
            LET'S START
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
